CDCgov/prime-reportstream
# Introduce `etor-ti` topic to Universal Pipeline

This PR introduces a new topic for ETOR. This topic will be used for senders and receivers that are parts of our ETOR solution in collaboration with Flexion's Trusted Intermediary.

These changes do require some attention for deployment to staging/prod:
- Since messages now include a topic, any existing queued messages may lead to deserialization failures. I'll mention this to the team to strategize on our rollout, and maybe temporarily having a default setting for the topic would be enough to get through the short-term
- The existing flexion organization settings in staging will need to be updated

Test Steps:
1. Reload updated etor settings with this command `./prime multiple-settings set --input ./settings/staging/0149-etor.yml`
2. Send a lab order report using this postman collection: 
[Temp Collection.postman_collection_json.txt](https://github.com/CDCgov/prime-reportstream/files/11548178/Temp.Collection.postman_collection_json.txt)

3. Query the submission history to verify that the report is sent and the topic is properly recorded, and check the sftp folder to look for the resulting HL7V2 file.

## Changes
- Updated org settings in `0149-etor.yml` to use new topic, and to shorten the organization's schema names (see #9576)
- In `SettingsProvider.kt` added a new topic for ETOR_TI, and added a property to the enum to indicate usage of the universal pipeline.
- In several files, updated existing checks like `topic == Topic.FULL_ELR` to use `topic.isUniversalPipeline` instead.
- In order for the submission history to accurately reflect the correct topic, updated Report.kt to take a topic value
- Updated `FullELRSender` to take a topic value instead of defaulting to FULL_ELR, also adjusted deserialization of the sender
- Updated tests to match new parameter requirements (explicitly using  `full-elr` for now)
- Added a Topic Binding and Converter so jooq can retrieve a topic `VARCHAR` from the database directly as a `Topic`

## Checklist

### Testing

## Linked Issues
- Addresses #8592 

## To Be Done
- In the course of this ticket, I discovered a bug and opened #9576
